src/planner/expression_binder/order_binder.cpp

```diff
@@ -100,6 +101,31 @@ unique_ptr<Expression> OrderBinder::Bind(unique_ptr<ParsedExpression> expr) {
 	case ExpressionClass::PARAMETER: {
 		throw ParameterNotAllowedException("Parameter not supported in ORDER BY clause");
 	}
+	case ExpressionClass::COLLATE: {
+		auto &collation = expr->Cast<CollateExpression>();
+		if (collation.child->expression_class == ExpressionClass::CONSTANT) {
+			auto &constant = collation.child->Cast<ConstantExpression>();
+			auto index = NumericCast<idx_t>(constant.value.GetValue<idx_t>()) - 1;
+			if (index >= extra_list->size()) {
+				throw BinderException("ORDER term out of range - should be between 1 and %lld", (idx_t)max_count);
+			}
+			auto &sel_entry = extra_list->at(index);
+			if (!sel_entry->alias.empty()) {
+				vector<string> column_names = {sel_entry->alias};
+				auto colref = make_uniq<ColumnRefExpression>(std::move(column_names));
```

Comment:
I don't quite follow what the goal of this code path is, e.g. it seems like this would not work:

```sql
SELECT 'a' AS my_alias ORDER BY 1 COLLATE NOCASE;
```

Perhaps we can just remove this code path and always use copy?